synapse/handlers/federation.py

```diff
@@ -2828,6 +2835,12 @@ async def _check_signature(self, event, context):
                         return
             except Exception as e:
                 last_exception = e
+
+        if last_exception is None:
+            # we can only get here if get_public_keys() returned an empty list
+            # TODO: make this better
+            raise RuntimeError("no public key in invite event")
```

Comment:
Seems fine, but curious if we were actually running into it?

Reply:
I don't *think* so, but I had to do something here to keep mypy happy.

Reply:
in theory I suspect we could run into it if the invite was malformed.

Reply:
Makes sense! The result is similar either way, wasn't sure if you came across it in Sentry or something. 👍